td-shim/src/acpi.rs

```diff
@@ -105,17 +105,13 @@ impl Xsdt {
 
     pub fn add_table(&mut self, addr: u64) {
         if self.header.length < size_of::<GenericSdtHeader>() as u32 {
-            log::error!(
-                "Invalid header: Xsdt header length should not be less than generic header size"
-            );
+            return;
```

Comment:
If this is expected, then we need return error.

If this is unexpected, then we need use assert.

Reply:
if you have another issue to resolve, please add issue number here.